compilation fail on snow leopard

Hey,

When I try to npm install gumbo-parser it seems to get to the point of compiling the parser itself in C, but blows up with the following:

  CC(target) Release/obj.target/gumbo/deps/gumbo-parser/src/attribute.o
../deps/gumbo-parser/src/attribute.c: In function ‘gumbo_get_attribute’:
../deps/gumbo-parser/src/attribute.c:30: error: ‘for’ loop initial declaration used outside C99 mode
../deps/gumbo-parser/src/attribute.c:30: warning: comparison between signed and unsigned
make: **\* [Release/obj.target/gumbo/deps/gumbo-parser/src/attribute.o] Error 1

I'm sure this is something to do with the GYP build script for the parser, but after trying for quite a while and to no avail to patch it I think it's about time to ask. My guess is the gnu99 requirement being enforced isn't correct, but even if I'm on the right track I can't figure out how to relax that requirement.

Any suggestion would be most welcome.

Thanks, Alex J Burke.

--- a/deps/gumbo-parser/gumbo-parser.gyp
+++ b/deps/gumbo-parser/gumbo-parser.gyp
@@ -35,6 +35,11 @@
        'target_name': 'gumbo',
        'product_prefix': 'lib',
        'type': 'static_library',
+       'xcode_settings': {
+          'OTHER_CFLAGS': [
+            '-Wall', '-Wno-sign-compare', '-std=gnu99'
+          ],
+        },
        'sources': [
             'src/attribute.c',
             'src/attribute.h',

